iosxr_iso2vbox.py

```diff
@@ -69,29 +69,54 @@
 import argparse
 from argparse import RawDescriptionHelpFormatter
 import re
+import logging
 
 # Telnet ports used to access IOS XR via socat
 console_port = 65000
 aux_port = 65001
 
 
-def run(command_string, debug=False):
+def run_process_quietly(cmd_string, debug=False):
     '''
-    Execute a CLI command string and return the result.
+    Execute a CLI command in the 'background' as process.
+    '''
+    argv = shlex.split(cmd_string)
+    if debug is True:
+        print ('argv: %s' % argv)
+    with open(os.devnull, 'w') as fp:
+        subprocess.Popen((argv), stdout=fp)
+    time.sleep(2)
 
-    Note this does not handle operators which need whitespace.
 
+def run_get_output(cmd_string, debug=False):
+    '''
+    Execute a CLI command string and return the output.
+    Note this does not handle operators which need whitespace.
     In those cases the raw commands will need to be used.
-
     E.g. passing grep 'a string separated by spaces'.
     '''
-    argv = command_string.split()
+    argv = cmd_string.split()
     if debug is True:
         print ('argv: %s' % argv)
     result = subprocess.check_output(argv)
     return result
 
 
+def run(cmd_string, debug=False):
+    '''
+    Execute a CLI command string quietly and return the result.
+    Note this does not handle operators which need whitespace.
+    In those cases the raw commands will need to be used.
+    E.g. passing grep 'a string separated by spaces'.
+    This is the bash equivalent of command > /dev/null.
+    '''
+    argv = cmd_string.split()
+    if debug is True:
+        print ('argv: %s' % argv)
+    FNULL = open(os.devnull, 'w')
```

Comment:
Would be nice to have it take a list as input rather than a string so you don't have to do the split and "guess" what the user really wants. You could then do:

```python
def run(args):
...

run('ssh-keygen -R [localhost]:2222'.split())
```

but frankly I'd be just as happy with:

```python
run(['ssh-keygen', '-R', '[localhost]:2222'])
```

as it avoids any possible confusion about whitespace.